test/DistributionTest.js

```diff
@@ -0,0 +1,151 @@
+            expectEvent(_beneficiaries, "RewardCollect", { beneficiaries_: acc3, amount_: ether('110'), transferStatus_: true });
+            await expect(beneficiaries.length).to.be.equal(amounts.length)
+        });
+
+    });
+
+    describe("Add beneficiary function", async () => {
+
+        it("Add beneficiary - reverted , address = 0", async () => {
+            let addr = constants.ZERO_ADDRESS;
+            await expectRevert(instanceDistribution.addBeneficiary(addr, ether('30')), "beneficiary or amount equal to 0");
+        });
+
+        it("Add beneficiary - reverted , amount = 0", async () => {
+            await expectRevert(instanceDistribution.addBeneficiary(acc4, new BN(0)), "beneficiary or amount equal to 0");
+        });
+
+        it("Add beneficiary - reverted , caller is not owner", async () => {
+            await expectRevert(instanceDistribution.addBeneficiary(acc4, new BN(0), { from: acc2 }), "caller is not the owner");
+        });
+
+        it("Add beneficiary - success", async () => {
+            let beneficiar = await instanceDistribution.addBeneficiary(acc4, ether('30'));
+            expectEvent(beneficiar, "RewardCollect", { beneficiaries_: acc4, amount_: ether('30'), transferStatus_: true });
+            expect(await instanceDistribution.balanceOfBeneficiares(acc4)).to.be.bignumber.equal(ether('30'));
+        });
+
+    });
+
+    describe("Decrease reward function", async () => {
+
+        it("Decrease reward - must reverted , amount = 0", async () => {
+            await expectRevert(instanceDistribution.decreaseReward(acc3, new BN(0)), "beneficiary or amount equal to 0");
+        });
+
+        it("Decrease reward - must reverted , address = 0", async () => {
+            let addr = constants.ZERO_ADDRESS;
+            await expectRevert(instanceDistribution.decreaseReward(addr, ether('30')), "beneficiary or amount equal to 0");
+        });
+
+        it("Decrease reward - must reverted , caller is not the owner ", async () => {
+            await expectRevert(instanceDistribution.decreaseReward(acc3, ether('30'), { from: acc2 }), "caller is not the owner");
+        });
+
+        it("Decrease reward - must success", async () => {
+            await instanceDistribution.decreaseReward(acc3, ether('30'));
+            expect(await instanceDistribution.balanceOfBeneficiares(acc3)).to.be.bignumber.equal(ether('80'));
+        });
+    });
+
+    describe("withdraw function", async () => {
+
+        it("Emergency withdraw - call from not owner reverted", async () => {
+            await expectRevert(instanceDistribution.emergencyWithdraw(ether('200'), { from: acc2 }), "caller is not the owner");
+        });
+
+        it("Emergency withdraw", async () => {
+            await instanceDistribution.emergencyWithdraw(ether('200'));
+            expect(await instanceToken.balanceOf(owner)).to.be.bignumber.equal(ether('800'));
+        });
+
+    });
+
```

Comment:
you skip tests for lockRewards function, who can call this function and other...